### PR TITLE
DatePicker: Copy DatePicker code from OUFR to react-next

### PR DESCRIPTION
**Pull request by MLoughry**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

#### Description of changes

Copy `DatePicker` component code to `react-next` as basis for migration.

### CODE REVIEW

**Comment by MLoughry** [comment]
@dzearing @ecraig12345 As I mentioned, I'll be on mini-paternity leave for the next two weeks. If this is merged, the migration changes for `DatePicker` are in [this branch](https://github.com/MLoughry/office-ui-fabric-react/tree/functional/date-picker)

**Comment by ecraig12345** [comment]
@lorejoh12 If we're converting the DatePicker to a function component, should we be starting from the OUFR one or the date-time one? How different are the APIs? (I think we're trying to avoid breaking API changes in the react-next components for now.)

**Comment by lorejoh12** [comment]
> @lorejoh12 If we're converting the DatePicker to a function component, should we be starting from the OUFR one or the date-time one? How different are the APIs? (I think we're trying to avoid breaking API changes in the react-next components for now.)

I'd strongly suggest starting with the date-time one, since the OUFR one has bugs in it that have been fixed in the date-time one. That being said, there are very minor breaking API changes between them- specifically, a couple of deprecated props were removed from `ICalendarProps` type, which is the type of the `calendarProps` property.

Otherwise, I don't think there are large differences between the OUFR DatePicker and the date-time DatePicker- the biggest difference is which Calendar component it wraps.

**Comment by MLoughry** [comment]
@ecraig12345 @lorejoh12 What's the plan, then? Should we migrate `@uifabric/date-picker` in place without a major version bump?

File: packages/react-next/src/components/DatePicker/examples/DatePicker.Bounded.Example.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { DatePicker, DayOfWeek, IDatePickerStrings } from 'office-ui-fabric-react/lib/DatePicker';
```
Comment (xugao) [inline packages/react-next/src/components/DatePicker/examples/DatePicker.Bounded.Example.tsx:2]: pls fix import to be `@fluentui/react-next` in examples

**Comment by MLoughry** [comment]
What's the plan, then? Should I migrate @uifabric/date-time in-place (without a major version), or copy that one over to react-next?


-Michael

________________________________

**Pull request closed without merging**